src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -690,6 +690,16 @@ public CompilerFeatureRequiredAttribute(string featureName)
             }
             """;
 
+        internal const string CollectionBuilderAttributeDefinition = """
+            namespace System.Runtime.CompilerServices
+            {
+                public sealed class CollectionBuilderAttribute : Attribute
```

Comment:
Consider applying AttributeUsageAttribute matching the proposal so that the end user diagnostic experience is more clear.

Reply:
Added `[AttributeUsage(AttributeTargets.All, Inherited = false, AllowMultiple = false)]`. I don't know if it's necessary to limit `AttributeTargets` for the tests.